cli/cmd/deploy/deploy.go

```diff
@@ -0,0 +1,397 @@
+	// no input taken from user for below
+	region   string
+	dbDriver string
+	dbDSN    string
+	slots    int64
+}
+
+// DeployCmd is the guided tour for deploying rill projects to rill cloud.
+// TODO :: add non interactive mode
+func DeployCmd(cfg *config.Config) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Guided tour for deploying rill projects to rill cloud",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx := cmd.Context()
+			warn := color.New(color.Bold).Add(color.FgYellow)
+			info := color.New(color.Bold).Add(color.FgWhite)
+			success := color.New(color.Bold).Add(color.FgGreen)
+			var adminClient *client.Client
+
+			// log in if not logged in
+			if !cfg.IsAuthenticated() {
+				msg := fmt.Sprintf("In order to deploy to Rill Cloud, you must login. Opening your browsers to %s to login or sign up...", cfg.AdminURL)
+				warn.Println(msg)
+				// NOTE : calling commands within commands has both pros and cons
+				// PRO : No duplicated code
+				// CON : Need to make sure that UX under sub command verifies with UX on this command
+				loginCmd := auth.LoginCmd(cfg)
+				loginCmd.SetContext(ctx)
+				if err := loginCmd.RunE(loginCmd, nil); err != nil {
+					exitWithFailure(err)
+				}
+
+				// set token in config
+				token, err := dotrill.GetAccessToken()
+				if err != nil {
+					return fmt.Errorf("could not parse access token from ~/.rill: %w", err)
+				}
+				cfg.AdminTokenDefault = token
+
+				// init admin client
+				adminClient, err = cmdutil.Client(cfg)
+				if err != nil {
+					return err
+				}
+			} else {
+				// switch is already part of login cmd so running this only when user is already logged in
+				defaultOrg, err := dotrill.GetDefaultOrg()
+				if err != nil {
+					return err
+				}
+
+				// init admin client
+				adminClient, err = cmdutil.Client(cfg)
+				if err != nil {
+					return err
+				}
+
+				multipleOrgs, err := multipleOrgs(adminClient)
+				if err != nil {
+					return fmt.Errorf("listing orgs failed with error %w", err)
+				}
+
+				if multipleOrgs {
+					msg := fmt.Sprintf("This project will be deployed under %s. Press Y to confirm and N to select a different org", defaultOrg)
+					if !cmdutil.ConfirmPrompt(msg, true) {
+						switchCmd := org.SwitchCmd(cfg)
+						switchCmd.SetContext(ctx)
+						if err := switchCmd.RunE(switchCmd, nil); err != nil {
+							exitWithFailure(err)
+						}
+					}
+				}
+			}
+
+			org, err := dotrill.GetDefaultOrg()
+			if err != nil {
+				return err
+			}
+
+			path, err := os.Getwd()
+			if err != nil {
+				return err
+			}
+
+			// verify current directory has rill project
+			if !hasRillProject(path) {
+				warn.Printf("\nCurrent path %s doesn't have a valid rill project. Please select correct path.\n", path)
+				warn.Printf("In case there is no valid rill project present, Please use `rill init` to create an empty rill project.\n\n")
+				// project path prompt
+				path = projectPathPrompt()
+			}
```

Comment:
I worry this opens up too much unintended possible behavior. In other commands, you can pass `--project` to pass the path to your project. How about just erroring and saying people can try again passing `--project` if it's in another directory? Then people can also leverage their terminal's path auto-complete, which will be better

Reply:
Here the idea was to error out as less as possible so that user doesn't come out of the interactive mode. 
We can add `--project` parameter as well which would override this behaviour and would error out if its not a valid project. What do you think ?

Reply:
This is one case where I think it's worth erroring out. Way most people will do `rill deploy` from their project folder. I don't trust our ability to create a good/safe file browser compared to the auto-complete of the terminal.